bayes_kit/hmc.py

```diff
@@ -23,7 +23,11 @@ def __init__(
         self._steps = steps
         self._metric = metric_diag or np.ones(self._dim)
         self._rng = np.random.default_rng(seed)
-        self._theta = init if init is not None else self._rng.normal(size=self._dim)
+        self._theta = (
+            init
+            if (init is not None and init.shape != (0,))
```

Comment:
Note that you can easily construct an empty numpy array whose shape will not be `(0,)`:

```python
>>> np.zeros((0,0))
array([], shape=(0, 0), dtype=float64)
```

The most general way to do this is to check the `.size` attribute is nonzero, but then you might as well check that the `.size` attribute is actually the correct size for the model